### PR TITLE
Connect resources and population to game state

### LINKED ISSUE #66: Connect resources and population to game state

### PR DESCRIPTION
**Pull request by bartoszbialecki**

### CODE REVIEW

**Review by pdybowski — APPROVED** [review]
conflict

File: src/app/screens/gamepage/index.ts [hunk 1]
```diff
@@ -46,7 +52,21 @@ export class GamePage {
     ) as HTMLButtonElement
     this.gameScreen = document.querySelector('.game__screen') as HTMLElement
     this.unitsImg = document.querySelector('.units__image') as HTMLDivElement
-    this.progressBar = document.querySelector(".progress-bar") as HTMLElement
+
+    this.goldAmountElement = document.querySelector(
+      '#goldAmount'
+    ) as HTMLElement
```
Comment (pdybowski) [inline src/app/screens/gamepage/index.ts:61]: We can set it to HtmlSpanElement - to be precisely

File: src/app/screens/gamepage/index.ts [hunk 2]
```diff
@@ -46,7 +52,21 @@ export class GamePage {
...
+      '#woodAmount'
+    ) as HTMLElement
+    this.stoneAmountElement = document.querySelector(
+      '#stoneAmount'
+    ) as HTMLElement
+    this.populationElement = document.querySelector(
+      '.population__count'
+    ) as HTMLElement
+
+    this.progressBar = document.querySelector('.progress-bar') as HTMLElement
```
Comment (pdybowski) [inline src/app/screens/gamepage/index.ts:76]: HtmlDivElement

File: src/app/screens/gamepage/index.ts [hunk 3]
```diff
@@ -75,17 +95,36 @@ export class GamePage {
...
+  private updateStoneAmount() {
+    this.stoneAmountElement.textContent = this.game.getStoneAmount().toString()
+  }
+
+  private updatePopulation() {
+    this.populationElement.textContent = this.game.getPopulation().toString()
+  }
+
+  private updateNextAttackProgressBar() {
+    const width = (this.game.getNextAttack() * 100) / 30
```
Comment (pdybowski) [inline src/app/screens/gamepage/index.ts:140]: Maybe we can use here calculateNextAttack method from Game.ts

**Review by pdybowski — APPROVED** [review]
Comments, optional

File: src/app/screens/gamepage/index.ts (hunk 1 above)
Comment (bartoszbialecki) [reply src/app/screens/gamepage/index.ts:61]: Is there any point on doing this? Actually we don't use anything specific for span elements and when I want to change html then I have to go here and change it too. I think we should use abstractions when possible

File: src/app/screens/gamepage/index.ts (hunk 3 above)
Comment (bartoszbialecki) [reply src/app/screens/gamepage/index.ts:140]: what for? calculateNextAttack calculates only new countdown (in seconds) of next attack when current attack is finished.

**Review by pdybowski — APPROVED** [review]

**Review by pdybowski — APPROVED** [review]

File: src/app/game.ts [hunk 4]
```diff
@@ -23,6 +23,10 @@ import {
   Resource,
   Fractions,
 } from './models'
+import { Barracks } from './models/buildings/barracks'
+import { Quarry } from './models/buildings/quarry'
```
Comment (pdybowski) [inline src/app/game.ts]: noo, just export from models

**Review by pdybowski — CHANGES_REQUESTED** [review]

File: src/app/game.ts (hunk 4 above)
Comment (bartoszbialecki) [reply src/app/game.ts]: yeah, I made it too fast and I've forgotten

**Review by hoolek77 — APPROVED** [review]

**Review by pdybowski — APPROVED** [review]

**Pull request merged**